src/lib/storage/chunk.cpp

```diff
@@ -197,7 +197,8 @@ std::vector<std::shared_ptr<const AbstractSegment>> Chunk::_get_segments_for_ids
                   if (column_id >= static_cast<ColumnID>(column_count())) return false;
                 return true;
               }()),
-              "column ids not within range [0, column_count()).");
+              "IndexColumnID " + std::to_string(column_ids[column_ids.size() - 1]) +
```

Comment:
We don't have a data type `IndexColumnID` --> `ColumnID`.
Further, it is `column_id` which exceeds the maximal column index.
`exceeds maximal column index` --> `exceeds the maximum column index`.